### PR TITLE
PYTHON-3075 [v4.0] bulk_write does not apply CodecOptions to upserted_ids result

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by ShaneHarvey** [comment]
evergreen patch

**Comment by ShaneHarvey** [comment]
evergreen patch

File: pymongo/message.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
                   _decode_selective,
                   _dict_to_bson,
                   _make_c_string)
+from bson import codec_options
```
Comment (ShaneHarvey) [inline pymongo/message.py:35]: Unneeded.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

**Comment by ShaneHarvey** [comment]
(updated the title because Backport-4 was being interpreted as a link to a jira ticket).

File: pymongo/message.py (hunk 1 above)
Comment (blink1073) [reply pymongo/message.py]: Fixed

**Comment by blink1073** [comment]
Encryption run [passed](https://spruce.mongodb.com/task/mongo_python_driver_4.0_tests_python_version_amazon1_test_encryption__platform~awslinux_auth_ssl~noauth_nossl_python_version~3.8_encryption~encryption_test_4.4_sharded_cluster_patch_93334cac4359c23fe75c3cb2062d104944718435_61fb13512fbabe23afff190b_22_02_02_23_27_14/tests?execution=0&sortBy=STATUS&sortDir=ASC).

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**